pkg/kubelet/volumemanager/reconciler/reconstruct_common.go

```diff
@@ -236,17 +236,25 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (rvolume *reconstructe
 	// Searching by spec checks whether the volume is actually attachable
 	// (i.e. has a PV) whereas searching by plugin name can only tell whether
 	// the plugin supports attachable volumes.
-	attachablePlugin, err := rc.volumePluginMgr.FindAttachablePluginBySpec(volumeSpec)
-	if err != nil {
-		return nil, err
-	}
 	deviceMountablePlugin, err := rc.volumePluginMgr.FindDeviceMountablePluginBySpec(volumeSpec)
 	if err != nil {
 		return nil, err
 	}
+	needsNameFromSpec := deviceMountablePlugin != nil
```

Comment:
Can we document why we need name from spec if volume plugin is attachable or mountable?

Reply:
I copied comment from DSWP: https://github.com/kubernetes/kubernetes/blob/1f2d49972cc541606b7a2a902271aa293b836d17/pkg/kubelet/volumemanager/cache/desired_state_of_world.go#L279-L283
The unique volume name must be reconstructed the same way.